### PR TITLE
[backbone] Documentation for backbone

### PR DESCRIPTION
**Pull request by kparichay**
Added documentation for backbone in ini-configuration

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: docs/configuration-ini.md [hunk 1]
```diff
@@ -255,6 +255,39 @@ bias_initializer = zeros
 Activation = softmax
 ```
 
+### Backbone section
+
+Describe another model, termed as backbone, to be used used in the model described by the current ini file. The backbone to be used can described with another ini configuration file path, or with model file for external frameworks. Support for backbones of external framework for Tensorflow-Lite is provided natively with Tensorflow-Lite framework. Support for backbones of other external frameworks is done using nnstreamer and its plugin. When using nnstreamer for external framework, ensure to add the corresponding baseline ML framework and its corresponding nnstreamer plugin as a dependency or install manually. For example, when using PyTorch based model as a backbone, both the packages *PyTorch* and *nnstreamer-pytorch*.
```
Comment (zhoonit) [inline docs/configuration-ini.md:260]: Could you have one sentence in a line? Github doesn't allow me to review part by part :(

Comment (kparichay) [reply docs/configuration-ini.md:260]: I will update.

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: docs/configuration-ini.md (hunk 1 above)
Comment (kparichay) [reply docs/configuration-ini.md]: done.

File: docs/configuration-ini.md [hunk 2]
```diff
@@ -255,6 +255,48 @@ bias_initializer = zeros
 Activation = softmax
 ```
 
+### Backbone section
+
+Describe another model, termed as backbone, to be used used in the model described by the current ini file.
```
Comment (zhoonit) [inline docs/configuration-ini.md:260]: Maybe `Backbone section describes ... `?? and `used` is used twice. `to be used used`,

File: docs/configuration-ini.md [hunk 3]
```diff
@@ -255,6 +255,48 @@ bias_initializer = zeros
...
+The backbone to be used can described with another ini configuration file path, or with model file for external frameworks.
```
Comment (zhoonit) [inline docs/configuration-ini.md:261]: `can be described` ??

File: docs/configuration-ini.md [hunk 4]
```diff
@@ -255,6 +255,48 @@ bias_initializer = zeros
...
+Support for backbones of external framework for Tensorflow-Lite is provided natively with Tensorflow-Lite framework.
+Support for backbones of other external frameworks is done using nnstreamer and its plugin.
+When using nnstreamer for external framework, ensure to add the corresponding baseline ML framework and its corresponding nnstreamer plugin as a dependency or install manually.
+For example, when using PyTorch based model as a backbone, both the packages *PyTorch* and *nnstreamer-pytorch*.
```
Comment (zhoonit) [inline docs/configuration-ini.md:265]:
```suggestion
For example, when using PyTorch based model as a backbone, both the packages *PyTorch* and *nnstreamer-pytorch* should be installed.
```

File: docs/configuration-ini.md [hunk 5]
```diff
@@ -255,6 +255,48 @@ bias_initializer = zeros
...
+
+Backbones made of nntrainer models, described using ini, support training the backbone also.
+However, this is not supported with external frameworks.
+It is possible to describe a backbone inside a backbone ini configuration file, as well as listing down multiple backbones to build a single model.
+For backbone ini configuration file, Model and Dataset sections are ignored.
+
+Describing a backbone is very similar to describing a layer.
+Start with a "[ ${layer name} ]" which must be unique throughtout the model. In case of backbone, the name of the backbone is preprended to the name of all the layers inside the backbone.
```
Comment (zhoonit) [inline docs/configuration-ini.md:273]: typo: Prepended

File: docs/configuration-ini.md [hunk 6]
```diff
@@ -255,6 +255,48 @@ bias_initializer = zeros
...
+    * .pb / .pt / .py / .circle etc via NNStreamer (corresponding nnstreamer plugin required)
+
+2. ```trainable = <bool>```
+
+   If this backbone must be trained. Only supported for ini backbones (nntrainer models).
+``
+Below is sample backbone section.
+
+```ini
+# block1
```
Comment (zhoonit) [inline docs/configuration-ini.md:289]: Could you add the omitted Model section and other simple layers? I think it would be hard to understand where this goes.

**Review by zhoonit — COMMENTED** [review]
I am very bad at writing so please ignore if you don't feel like it :p

File: docs/configuration-ini.md (hunk 6 above)
Comment (kparichay) [reply docs/configuration-ini.md:293]: done.

File: docs/configuration-ini.md (hunk 5 above)
Comment (kparichay) [reply docs/configuration-ini.md]: done.

File: docs/configuration-ini.md (hunk 4 above)
Comment (kparichay) [reply docs/configuration-ini.md]: done.

File: docs/configuration-ini.md (hunk 3 above)
Comment (kparichay) [reply docs/configuration-ini.md]: done.

File: docs/configuration-ini.md (hunk 2 above)
Comment (kparichay) [reply docs/configuration-ini.md]: done.

**Review by zhoonit — APPROVED** [review]
LGTM
